pkg/lwip/include/arch/sys_arch.h

```diff
@@ -34,29 +35,69 @@
 extern "C" {
 #endif
 
+/**
+ * @brief   System configuration
+ */
+/* prefer mutexes rather than binary semaphores */
 #define LWIP_COMPAT_MUTEX   (0)
-#define SYS_SEM_NULL        { 0, PRIORITY_QUEUE_INIT }
-#define SYS_MBOX_SIZE       (8)
+/** @} */
 
-typedef struct {
-    mbox_t mbox;
-    msg_t msgs[SYS_MBOX_SIZE];
-} sys_mbox_t;
+/**
+ * @name    Critical sections protection definitions
+ * @see     https://www.nongnu.org/lwip/2_1_x/group__sys__prot.html
+ * @{
+ */
+#define SYS_ARCH_PROTECT(x)         mutex_lock(&x)
+#define SYS_ARCH_UNPROTECT(x)       mutex_unlock(&x)
+#define SYS_ARCH_DECL_PROTECT(x)    mutex_t x = MUTEX_INIT
+/** @} */
 
-typedef mutex_t sys_mutex_t;
-typedef sema_t sys_sem_t;
-typedef kernel_pid_t sys_thread_t;
+/**
+ * @name    Semaphors definitions
```

Comment:
Typo here: s/Semaphors/Semaphore/

This is my last comment, you can squash and we are good.

Reply:
Following the schema of the rest of this file (taking the titles from the lwIP doc and appending "definitions") it should be "Semaphores", but yes will do :-)